tcf_website/views/search.py

```diff
@@ -154,3 +184,50 @@ def normalize_search_query(q: str) -> str:
     ]
 
     return courses
+
+
+def filter_courses(filters):
```

Comment:
Could also consider combining these methods somehow, as having two filter blocks of code to maintain seems redundant. Perhaps extract the filter/query DB results to separate functions that are called on the Course.objects?

Reply:
You could make the fetch_courses(query) handle the queries for both a valid search query and empty "" queries, with an if statement for the results, with a call to a general "filters" function that applies the 4 filters currently.